internal/api/v1alpha1/authproxyworkload_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	cloudsqlapi "github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAuthProxyWorkload_ValidateCreateUpdate(t *testing.T) {
+	type testcase struct {
+		desc            string
+		spec            cloudsqlapi.AuthProxyWorkloadSpec
+		oldSpec         *cloudsqlapi.AuthProxyWorkloadSpec
+		wantCreateValid bool
```

Comment:
Could we split create tests out from update tests? It might make the tests easier to read (and write)?

Reply:
Fixed.